Feature/reply/exception Reply 예외처리 Entity 내부에서 수행하도록 수정

JangCoding opened the pull request:
Reply Entity의 @NoBlank 어노테이션을 제거하고,
Reply 내부에서 파라미터 값을 확인하여 에러처리를 할 수 있도록 변경하였습니다.

진행 과정에서 BaseEntity를 상속받도록 수정하였고
Service 에서 처리되던 내용을 Reply Entity 로 이동하였습니다.

close #19

JangCoding committed:
schema.sql 에 Reply 테이블 설정

JangCoding committed:
Reply Entity수정 , 예외처리 위치 변경
- BaseEntity 상속 받도록
- 변수명 통일 ( createdDateTime, LastModifiedDateTime)
- DateTime 값 BaseEntity에서 처리 되도록
- 예외처리 Entity 내부에서 처리할 수 있도록

JangCoding committed:
Reply Repository, DTO 수정
- Repositoy 메서드명 수정
- Request @NotBlank 어노테이션 삭제

rugii913 commented on src/main/kotlin/com/bee/newsfeed_bee/domain/reply/dto/ReplyCreateRequest.kt:
```diff
@@ -3,7 +3,10 @@ package com.bee.newsfeed_bee.domain.reply.dto
-    @field:NotBlank(message = "userName은 필수 입력 정보입니다.") val userName: String,
-    @field:NotBlank(message = "password는 필수 입력 정보입니다.") val password: String,
-    @field:NotBlank(message = "contents는 필수 입력 정보입니다.") val contents: String
+    //@field:NotBlank(message = "userName은 필수 입력 정보입니다.")
```
보통 String의 NotBlank validation과 같은 처리는 web layer에서 많이 하는 듯해요.
- String 자체가 유효한 String인지("" 빈 문자열 금지, "   " 공백만 있는 문자열 금지)는 web layer에서 예외처리, 이 값이 우리 service에 적합한지(ex. 길이 ~자 이내)는 service에서 예외처리
- 판단은 다를 수 있고 팀에서 정하기 나름인 듯하지만 아마도?...
- 그런 얘기를 들었었습니다.

rugii913 replied:
여러 자료를 읽어보고 있습니다.
- <https://starkying.tistory.com/entry/Model-%EA%B0%9D%EC%B2%B4%EC%97%90%EC%84%9C%EB%A7%8C%ED%81%BC%EC%9D%80-Validation%EC%9D%84-%ED%95%84%EC%88%98%EB%A1%9C-%ED%95%98%EC%9E%90>
- <https://seongwon.dev/Spring-MVC/20220622-Valid%EB%9E%80/#7-dto%EC%97%90%EC%84%9C%EB%8A%94-valid%EB%A1%9C-%EC%96%B4%EB%94%94%EA%B9%8C%EC%A7%80-%EA%B2%80%EC%A6%9D%ED%95%B4%EC%95%BC%ED%95%A0%EA%B9%8C>
- <https://murphymoon.tistory.com/entry/%EC%9A%B0%EC%95%84%ED%95%9C-%ED%85%8C%ED%81%AC-MVC-%EB%A6%AC%EB%B7%B0-%EB%A0%88%EC%9D%B4%EC%96%B4-MVC-%ED%8C%A8%ED%84%B4-5%EB%A0%88%EC%9D%B4%EC%96%B4>
- <https://luvstudy.tistory.com/183>
- <https://velog.io/@rmswjdtn/Spring-Validation은-어디서-해야할까>
- <https://blog.barogo.io/개발인턴-validation의-적절성-79ed23f28a72>
- <https://heesangstudynote.tistory.com/109>

JangCoding replied:
아하.. 그렇게 또 분류가 나뉘는군요

rugii913 commented on src/main/kotlin/com/bee/newsfeed_bee/domain/reply/entity/Reply.kt:
```diff
@@ -11,41 +14,63 @@ import java.time.OffsetDateTime
+
+    @Column(name="user_name")
+    var userName = _userName
+        .also{if(it.isEmpty() || it.length > USERNAME_MAX)
```
저도 이렇게 also를 활용해봐야겠습니다

JangCoding committed:
예외 검증 처리 위치 회의 후 수정
- controller에서 데이터를 받아올 때 와 domain 에서 값을 처리할 때 둘 다 검증하는 것이 안정적일 것으로 판단하여 @Valid 어노테이션을 사용하는 것을 양립하기로 결정하였습니다.
